src/runway/path.py

```diff
@@ -0,0 +1,119 @@
+""" Runway configuration 'path' settings """
+import os
+import logging
+import six
+
+from .sources.git import Git
+
+LOGGER = logging.getLogger('runway')
+
+
+class Path(object):
+    """ Runway configuration 'path' settings object """
+
+    def __init__(self, module, env_root, cache_dir=None):
+        """ Initialize. """
+        if not cache_dir:
+            cache_dir = os.path.expanduser("~/.runway_cache")
+
+        self.env_root = env_root
+        self.cache_dir = cache_dir
+        self.source, self.uri, self.location, self.options = self.parse(module)
+        self.module_root = self.__get_module_root_dir(module)
+
+    def parse(self, module):
+        """ Retrieve the source and location of the path variable. """
+        source = 'local'
+        uri = ''
+        location = ''
+        options = ''
+
+        split_source_location = module.get('path').split('::')
+
+        # Local path
+        if len(split_source_location) != 2:
+            location = split_source_location[0]
+            return [source, uri, location, options]
+
+        source = split_source_location[0]
+        temp_location = split_source_location[1]
+
+        uri, location = self.__parse_uri_and_location(temp_location)
+        location, options = self.__parse_location_and_options(location)
+
+        return [source, uri, location, options]
+
+    def configuration(self):
```

Comment:
This may be better served as an `@property`.

Reply:
So this one is a little weird. My pylint was giving up issues because I didn't have enough public methods (2) on the class. I converted this to appease that... but it seems like a really weird requirement. In the past I've used a single public method for a lot of Service objects in other languages such as `call()`. 

Is that a pylint requirement we should look to correct?

Reply:
And TIL about `@property` :D

Reply:
you can always add `#  pylint: disable=too-few-public-methods` to the same line as the class definition if the class is intended to only have 1.